src/releasers/dotnet.ts

```diff
@@ -0,0 +1,152 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import * as chalk from 'chalk';
+import {ReleasePR} from '../release-pr';
+
+import {ConventionalCommits} from '../conventional-commits';
+import {checkpoint, CheckpointType} from '../util/checkpoint';
+import {Update} from '../updaters/update';
+
+// Generic
+import {Changelog} from '../updaters/changelog';
+// Dotnet specific files
+import {CommonProperties} from '../updaters/dotnet/common-properties';
+import {Project} from '../updaters/dotnet/project';
+import {ReadMe} from '../updaters/dotnet/readme';
+
+export class DotNet extends ReleasePR {
+  static releaserName = 'dotnet';
+  // When release-please runs, it calls the method run() on the base
+  // ReleasePR class. run() performs operations common to all languages:
+  // checking if there is an unreleased release PR, checking if the release
+  // strategy supports snapshots.
+  //
+  // _run() is called from run(), with the expectation that it performs
+  // language specific release tasks, e.g., updating files specific to .NET.
+  protected async _run() {
+    if (this.snapshot) {
+      checkpoint('running as snapshot pre-release', CheckpointType.Success);
+    }
+    const latestTag = await this.gh.latestTag(
+      this.monorepoTags ? `${this.packageName}-` : undefined,
+      this.snapshot
+    );
+    const commits = await this.commits({
+      sha: latestTag ? latestTag.sha : undefined,
+      path: this.path,
+    });
+
+    const cc = new ConventionalCommits({
+      commits,
+      githubRepoUrl: this.repoUrl,
+      bumpMinorPreMajor: this.bumpMinorPreMajor,
+      changelogSections: this.changelogSections,
+    });
+    const candidate = await this.coerceReleaseCandidate(
+      cc,
+      latestTag,
+      this.snapshot
+    );
+
+    const changelogEntry = await cc.generateChangelogEntry({
+      version: candidate.version,
+      currentTag: `v${candidate.version}`,
+      previousTag: candidate.previousTag,
+    });
+
+    // don't create a release candidate until user facing changes
+    // (fix, feat, BREAKING CHANGE) have been made; a CHANGELOG that's
+    // one line is a good indicator that there were no interesting commits.
+    if (this.changelogEmpty(changelogEntry)) {
+      checkpoint(
+        `no user facing commits found since ${
+          latestTag ? latestTag.sha : 'beginning of time'
+        }`,
+        CheckpointType.Failure
+      );
+      return;
+    }
+
+    const updates: Update[] = [];
+
+    updates.push(
+      new Changelog({
+        path: 'docs/history.md',
+        changelogEntry,
+        version: candidate.version,
+        packageName: this.packageName,
+      })
+    );
+
+    updates.push(
+      new ReadMe({
+        path: 'README.md',
+        changelogEntry,
+        version: candidate.version,
+        packageName: this.packageName,
+      })
+    );
+
+    updates.push(
```

Comment:
Note that currently, this filename isn't common across all repositories. (This is one of the challenges of using something like release-please - we have slightly different requirements in all repos.)

This particular aspect could be standardized at least across functions-framework-dotnet and google-cloudevents-dotnet, but it's not like this is a convention across .NET in general. (We're in the relatively rare situation of having multiple packages to publish, all tied to the same version.)

Reply:
> This particular aspect could be standardized at least across functions-framework-dotnet and google-cloudevents-dotnet, but it's not like this is a convention across .NET in general.

Where we've started with other language teams is to start with a yoshi specific release process first, so that our team benefits. And, if there's a way to abstract it more generally for the community that's awesome.

@JustinBeckwith this is actually why I didn't document `dotnet` in the README, I've been trying to only document releasers that are generally useful here, i.e., not too tied to the yoshi team.

@jskeet one option is we could be explicit and call this a `yoshi-dotnet.ts` releaser, and try to just solve for your use case for now.